frontend-build-pipeline.yml

```diff
@@ -16,27 +19,48 @@ parameters:
 - name: sonarCloudProjectName # sonar cloud's cli project name
   type: string
   default: 'RevatureShopFrontend'
+- name: sonarCloudExclude # files for sonar cloud to ignore
+  type: string
+  default: '**/*.spec.ts'
+- name: sonarCloudCoverageDir
+  type: string
+  default: '$(Build.Repository.LocalPath)/coverage/lcov.info'
+- name: sonarCloudSources
+  type: string
+  default: '.'
 
 steps:
 
-
-
 # Node.js is required to build Angular apps
 - task: NodeTool@0
   inputs:
     versionSpec: '12.14'
   displayName: 'Install Node.js'
 
 - script: |
-    npm ci
+    ls $(Build.Repository.LocalPath)
+
+- script: |
+    npm install
+  displayName: 'npm Install'
+  workingDirectory: $(Build.Repository.LocalPath)
```

Comment:
you probably don't need these workingDirectory lines, isn't that already the default?
on the other hand, if you're parametrizing the other paths (like sonarCloudSources), that kind of suggests you want this pipeline to be able to handle the angular app being in a subdirectory (which is somewhat common) just by changing parameters. that would require a few changes, but it's not a big priority.

Reply:
Yeah you're right it's not really necessary to have those lines in my case. That would be a nice feature to handle the app being in different places. For right now I'll just remove the unnecessary lines.